compass/ocean/tests/mitgcm_baroclinic_gyre/initial_state.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+import xarray as xr
+from mpas_tools.io import write_netcdf
+
+from compass.ocean.vertical import init_vertical_coord
+from compass.step import Step
+
+
+class InitialState(Step):
+    """
+    A step for creating the initial condition for MITgcm baroclinic gyre
+    test cases
+
+    Attributes
+    ----------
+    resolution : str
+        The resolution of the test case
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        resolution : str
+            The resolution of the test case
+        """
+        super().__init__(test_case=test_case, name='initial_state')
+        self.resolution = resolution
+        self.add_input_file(
+            filename='culled_mesh.nc',
+            target='../cull_mesh/culled_mesh.nc')
+        self.add_input_file(
+            filename='culled_graph.info',
+            target='../cull_mesh/culled_graph.info')
+
+        self.add_output_file('initial_state.nc')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        config = self.config
+
+        dsMesh = xr.open_dataset('culled_mesh.nc')
+
+        ds = _write_initial_state(config, dsMesh)
+
+        _write_forcing(config, ds.latCell)
+
+
+def _write_initial_state(config, dsMesh):
+    ds = dsMesh.copy()
+
+    bottom_depth = config.getfloat('vertical_grid', 'bottom_depth')
+    ds['bottomDepth'] = bottom_depth * xr.ones_like(ds.xCell)
+    ds['ssh'] = xr.zeros_like(ds.xCell)
+
+    init_vertical_coord(config, ds)
+
+    # setting the initial conditions
+    temperature = (-11. * np.log(0.0414 *
+                   (-1. * ds.zMid + 100.3)) + 48.8)
+    temperature = temperature.transpose('Time', 'nCells', 'nVertLevels')
+    bottT = temperature[0, 200, -1].values
+    surfT = temperature[0, 200, 0].values
+    print(f'bottom T: {bottT} and surface : {surfT}')
+    salinity = 34.0 * xr.ones_like(temperature)
```

Comment:
Consider making certain initial temperature and salinity parameters a part of the `mitgcm_baroclinic_gyre` section in your config file

Reply:
The initial temperature is a function of depth.  @alicebarthel has constructed a function that's a pretty good approximation of the table of numbers used in the MITgcm version of the test case.  It's full of "magic" numbers so they wouldn't be super easy to add to the config file.  But maybe there would be a way to give the top (z=0) and bottom (z=bottom_depth) temperature and have them vary between those two with your prescribed functional form?  That would seem cleaner to me.

I wouldn't hardcode sampling `bottT` and `surfT` at the 200th grid cell here.  Either you should take grid cell 0 or you'll want to have a number somehow based on `nCells`.  We don't necessarily know that the mesh will have a 200th grid cell (though it's obviously highly likely) since we're flexible about the resolution.

Regarding the initial salinity, I agree for sure that that needs to be a config option rather than a magic number here.

Reply:
Good point @xylar re. cell 200. @cbegeman I like the idea of it being more general (setting T in cfg) but I have not come up with a satisfying way to do that at this stage. For the original purpose of matching MITgcm, hard-coding the structure was easiest. It can extend to a deeper set-up, but approximates their values over 0-1800m. I'll look to see about setting it from top and bottom T keeping the existing functional form. We could do a prettier functional form with, say, a thermocline depth parameter setting the change in T=f(z) shape, but I have not looked into it that closely.

Reply:
@alicebarthel, couldn't you have config options that are the temperatures in the middle of the top and bottom layers respectively?  Then, couldn't you take your existing functional form but stretch it linearly so it hits the desired top and bottom temperatures?

Alternatively, if you want to be very particular about having the same temperature profile independent of resolution, couldn't you do the same but instead of the middle of the top and bottom layers, you would use z = 0 and z=1800 m.  The default values would be chosen such that the temperature in the middle of the top and bottom layers is quite close to the value you currently get with your functional form.

Reply:
Moved the constant salinity value to the cfg file. The initial temperature profile is now set (still with a "magical" functional form and a fitted parameter) from the top and bottom temperature set at the surface (z=0m) and bottom (z = bottom_depth). The default values approximate the mitgcm stratification, but we could add a table to reach to be more exact.